### PR TITLE
Implement progress for centos

### PR DESCRIPTION
**Pull request by pengz1**
Jenkins depends on: https://github.com/RackHD/on-tasks/pull/370, https://github.com/RackHD/on-http/pull/523, https://github.com/RackHD/on-core/pull/227, https://github.com/RackHD/on-taskgraph/pull/189
Progress changes for CentOS installation.
For OS installation, we can't use a valid percentage, so change progress.percentage to an option value.

### CODE REVIEW

**Comment by pengz1** [comment]
@RackHD/corecommitters  @iceiilin @cgx027

**Comment by iceiilin** [comment]
:+1:

File: lib/workflow/stores/mongo.js [hunk 1]
```diff
@@ -994,7 +995,7 @@ function mongoStoreFactory(waterline, Promise, Constants, Errors, assert, _, san
     exports.updateGraphProgress = function(data) {
         assert.uuid(data.graphId, "Graph ID should be a UUID");
         assert.string(data.progress.description, "Progress info should include description");
```
Comment (cgx027) [inline lib/workflow/stores/mongo.js]: This is not in this PR, but maybe try ```assert.object(data.progress)``` before assert the members of data.progress. If data.progress is undefined, you will get ```TypeError: Cannot read property 'description' of undefined``` exception and the assertion will not work.

File: lib/workflow/stores/mongo.js [hunk 2]
```diff
@@ -372,6 +372,7 @@ function mongoStoreFactory(waterline, Promise, Constants, Errors, assert, _, san
         return waterline.graphobjects.findOne(query, options)
         .then(function(graph) {
             sanitizer.decrypt(graph.tasks[data.taskId]);
+            sanitizer.decrypt(graph.context);
```
Comment (yyscamper) [inline lib/workflow/stores/mongo.js]: I think you need rebase your code, this is @lanchongyizu 's commit, should not see it in your PR.

**Comment by yyscamper** [comment]
:+1: after rebasing PR.

**Comment by changev** [comment]
test this please

**Comment by changev** [comment]
test this please

**Comment by changev** [comment]
test this please

**Comment by changev** [comment]
test this please

**Comment by changev** [comment]
test this please

**Comment by yyscamper** [comment]
jenkins: test this please

**Comment by changev** [comment]
jenkins: test this please

**Comment by changev** [comment]
jenkins: test this please

**Comment by changev** [comment]
jenkins: test this please

**Comment by changev** [comment]
jenkins: test this please

**Pull request merged**